### ISSUE TITLE
Create Concepts page for data types

### ISSUE THREAD

**Issue by kgodey** [centerofci/mathesar-wiki#14]
We need to clearly define what a data type is and what a Mathesar type is.

Related discussions:
- https://github.com/centerofci/mathesar/discussions/838
- https://github.com/centerofci/mathesar/discussions/843
- https://github.com/centerofci/mathesar/discussions/849

**Comment by mathemancer** [comment]
Passing comment:

We have the following type concepts so far:
- proper PostgreSQL types that come out-of-the-box
- Mathesar types that are implemented as PostgreSQL `DOMAIN`s.  This means that they're just basic PostgreSQL types, plus a check condition that ensures they fit some form.  In the case of those currently implemented, they also come with DB-layer convenience functions to pluck out different parts (e.g., the local-part of an email) of the entry.
- Mathesar types that are implemented as display options of a different type (e.g., percents).  It's debatable whether these are proper types, since they don't really enforce anything or include any functionality beyond the underlying types.
- Composite types (we don't have these yet, but we've noted the concept in other places).  These are types composed of come combination of other primitive types.

Should all of the above look like "types" to the user?  If so, should the difference between a primitive type and `DOMAIN` type be exposed somehow?  (I've passed over the "function / generated" type above, since we already discussed that elsewhere).
